spec/enumerables_spec.rb

```diff
@@ -0,0 +1,273 @@
+require 'enumerables'
+
+RSpec.describe Enumerable do
+  let(:array1) { [1, 2, 3, 4] }
+  let(:array2) { %w[two three ten twelve] }
+  let(:array3) { [1, 4, nil, 8] }
+  let(:array4) { [false, nil, nil, false] }
+  let(:array5) { [1, 2, 3, 2, 5, 2] }
+  let(:range) { (1..5) }
+  describe '#my_each' do
+    it 'should raise error when argument given' do
+      expect { array1.my_each('a') }.to raise_error(ArgumentError)
+    end
+
+    context 'when block given' do
+      let(:new_array1) { [] }
+      let(:new_array2) { [] }
+
+      it 'should call block with one argument, the item (element), for each collection item' do
+        array2.my_each { |ele| new_array1.push(ele + '!') }
+        range.my_each { |ele| new_array2.push(ele + 2) }
+        expect(new_array1).to eq(%w[two! three! ten! twelve!])
+        expect(new_array2).to eq([3, 4, 5, 6, 7])
+      end
+    end
+
+    context 'when no block is given' do
+      it 'should return an Enumerator ' do
+        expect(array1.my_each).to be_a_kind_of(Enumerator)
+        expect(array1.my_each.to_a).to eq(array1)
```

Comment:
* [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

Your second expect is not related to the `it` statement you have. With just the first expect you would be checking for the enumerator, so you don't need it. Check that on all your other tests.